### ISSUE TITLE
Width problem with some float numbers

### ISSUE THREAD

**Issue by diegozea** [JuliaIO/Formatting.jl#51]
There is a problem with the width of the printed string when the rounded number has one digit more than the input number. e.g.:

```julia
julia> using Formatting

julia> function print_number(n)
       s = IOBuffer()
       printfmt(s, "{1:6.2f}", n)
       S = String(s)
       println("printed: \"", S, "\" number: ", n, " printed length: ", length(S), " expected: ", 6)
       end
print_number (generic function with 1 method)

julia> print_number(10.000)
printed: " 10.00" number: 10.0 printed length: 6 expected: 6

julia> print_number( 9.999)
printed: "  10.00" number: 9.999 printed length: 7 expected: 6

julia> print_number(100.000)
printed: "100.00" number: 100.0 printed length: 6 expected: 6

julia> print_number(99.999)
printed: " 100.00" number: 99.999 printed length: 7 expected: 6

```

This causes https://github.com/diegozea/MIToS.jl/issues/42

**Comment by jmkuhn** [comment]
I submitted PR #47 in January to fix this issue but it has not been merged.  I have a total of 5 submitted PRs for bug fixes going back 7 months.  None have been merged.  There are many other PRs that I'd like to submit but I have held back since none have been merged yet.  I will likely just start to work on a private fork to make progress.

**Comment by diegozea** [comment]
Maybe @ararslan or another JuliaIO member can add you to the organization to allow you maintain this package ;)

**Comment by ararslan** [comment]
I've added you as a collaborator on this repository, @jmkuhn. Thanks for your efforts and sorry for the lack of attention.

**Comment by jmkuhn** [comment]
I merged PR #47 into master which will resolve your issue.  I want to merge a few more PRs and get the 0.7 compat working again before I tag a new release.  Thanks @ararslan for allowing me to do this.

**Comment by diegozea** [comment]
Thanks @ararslan and @jmkuhn  :smiley:
